[FEATURE] Arch Linux installation script (create .pkg.tar.xz archive or install directly)

fandango- opened the pull request:
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used CCExtractor.
- [x] I have used CCExtractor just a couple of times.
- [ ] I absolutely love CCExtractor, but have not contributed previously.
- [ ] I am an active contributor to CCExtractor.

---

Add installation script for Arch Linux based distributions.

Steps to test this out (on an Arch Linux based distribution):
$ cd package_creators/
$ ./arch.sh

There is an option to do 2 things (install directly or create the .pkg.tar.xz archive for distribution purposes without installing on host machine).

Tested on
Kernel : Linux 4.4.48-1-MANJARO (x86_64)
C Library : GNU C Library version 2.24 (stable)
Default C Compiler : GNU C Compiler version 6.3.1 20170109 (GCC)
Distribution : Manjaro Linux

Resolves sub-part of Issue #678

rkuchumov commented:
1.  I've got `./arch.sh:read:8: -p: no coprocess`. (I fixed it by changing #!/bin/sh to bash, but you should check it)
2.  There may be no `sudo` in the system
3.  When ./autogen.sh or ./tarball.sh fails arch.sh still asks me to install the package.
4.  When I answer anything but `n` or `N` to the prompt, it still install the package. But default behavior everywhere in this case is to exit

fandango- commented:
Thanks for the review.
1. Changed to use bash as the shell
2. Use 'su' instead of 'sudo' now
3. Fixed; now if a command fails, the script won't continue
4. I was just emulating the default behaviour of pacman / makepkg - Y is the default there. However, I've now changed the prompt within arch.sh to have N as the default.

rkuchumov commented:
> Use 'su' instead of 'sudo' now

But when the system has sudo, you should use it :) Usually scripts try `sudo` at first, (which asks user password), and if they fail to find it, they use `su` (which asks root password). You should either implement this logic or just remove installation step and let user do it.

Don't forget to add `.gitignore`, to update `docs/CHANGES.TXT` and to add instructions to `README.md`

fandango- commented:
Done - 'sudo' is used if available; fallback to 'su'. Also, updated the mentioned files.

rkuchumov commented:
Ok, thanks.
It can be merged.
